### DISCUSSION TITLE
Upload compiled Arduino file from Bonsai

### DISCUSSION THREAD

**Discussion by mohebial** [bonsai-rx/bonsai#1320] [Q&A] [answered]
Hi,
    I am monitoring the mouse behavior using an Arduino board and want to use Bonsai for various operations. I was wondering if there is an easy way to upload a compiled Arduino code (.hex) on the board from Bonsai using the serial port. 
Thanks! 
Ali

**Comment by bruno-f-cruz** [comment, marked as answer]
Never uploaded an .hex file directly, but I have used the [`StartProcess`](https://bonsai-rx.org/docs/api/Bonsai.IO.StartProcess.html?q=startprocess) and the [Arduino CLI](https://arduino.github.io/arduino-cli/0.32/getting-started/) to upload scripts before.
